db/migrate/20230825233539_create_user_training_tip.rb

```diff
@@ -0,0 +1,15 @@
+class CreateUserTrainingTip < ActiveRecord::Migration[6.1]
+  def change
+    create_table :user_training_tips do |t|
+      t.references :user, null: false, foreign_key: true
+      t.references :tool, null: false, foreign_key: {to_table: :resources}
+      t.references :language, null: false, foreign_key: true
+      t.string :tip_id
+      t.boolean :is_completed
+
+      t.timestamps
+    end
+
+    add_index :user_training_tips, [:id, :tool_id, :language_id, :tip_id], unique: true, name: "training-tips-unique-index"
```

Comment:
I don't think you want `:id` in this unique index

Reply:
unless `:id` is the reference to the `:user` relationship which isn't currently present in this index

Reply:
Oh, yeah, it should be `user_id`